net/net.go

```diff
@@ -157,11 +155,12 @@ func (t *net) Store() lstore.Logstore {
 	return t.store
 }
 
-func (t *net) GetHostID(context.Context) (peer.ID, error) {
+func (t *net) GetHostID(_ context.Context) (peer.ID, error) {
 	return t.host.ID(), nil
 }
 
-func (t *net) CreateThread(_ context.Context, id thread.ID, opts ...core.KeyOption) (info thread.Info, err error) {
+func (t *net) CreateThread(_ context.Context, creds thread.Credentials, opts ...core.KeyOption) (info thread.Info, err error) {
```

Comment:
net/net.go is the layer where the ACL checks will eventually happen (not in this PR). Create/add thread can ensure that the caller is not locking themselves out of the thread.